CMakeLists.txt

```diff
@@ -120,7 +160,26 @@ else()
     message("-- INFO: R library not found. Not compiling R bindings.")
 endif()
 
-target_link_libraries(${CMAKE_PROJECT_NAME} OpenMP::OpenMP_CXX)
+# --- OpenMP handling: use MacPorts libomp on macOS, otherwise use OpenMP::OpenMP_CXX ---
+if(APPLE)
+    message(STATUS "Using MacPorts libomp for OpenMP on macOS")
+
+    # Compiler flags for AppleClang
+    set(OpenMP_CXX_FLAGS "-Xpreprocessor -fopenmp -I/opt/local/include -I/opt/local/include/libomp")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
+
+    # Linker flags pointing to MacPorts lib
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -L/opt/local/lib/libomp -lomp")
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -L/opt/local/lib/libomp -lomp")
+
+    # Also try to provide the variables that some CMake scripts check
+    set(OpenMP_CXX_LIB_NAMES "omp")
+    set(OpenMP_omp_LIBRARY "/opt/local/lib/libomp/libomp.dylib")
+else()
+    # normal systems: rely on find_package(OpenMP)
+    find_package(OpenMP REQUIRED)
+    target_link_libraries(${CMAKE_PROJECT_NAME} PRIVATE OpenMP::OpenMP_CXX)
```

Comment:
PRIVATE cannot be used. So line needs to be:
 target_link_libraries(${CMAKE_PROJECT_NAME} OpenMP::OpenMP_CXX)